iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -83,7 +83,8 @@ static const char* DEVICE_METHOD_RESPONSE_TOPIC = "$iothub/methods/res/%d/?$rid=
 
 static const char SYS_TOPIC_STRING_FORMAT[] = "%s%%24.%s=%s";
 
-static const char* REQUEST_ID_PROPERTY = "?$rid=";
+static const char REQUEST_ID_PROPERTY[] = "?$rid=";
+static size_t REQUEST_ID_PROPERTY_LEN = sizeof(REQUEST_ID_PROPERTY) - 1;
```

Comment:
Not necessary here, but I've been using the following macro:
```c 
#define lengthof(x) (sizeof(x) - 1)
```
Useful when we have several *_LEN defines in a single c module, since you reduce the number of defines to just one.

Reply:
Indeed!  Of all the macro utils we don't have a str_len one like that!  I don't see other places in file using it enough so will keep this as-is for now.